### PR TITLE
feature/16277-vbp-with-compare

### PR DESCRIPTION
**Pull request by raf18seb**
VBP indicator now follows the main series when the compare mode is on and [compareToMain](https://api.highcharts.com/highstock/series.vbp.compareToMain) is `false`. See #16277.

### CODE REVIEW

File: ts/Stock/Indicators/VBP/VBPIndicator.ts [hunk 1]
```diff
@@ -590,7 +591,19 @@ class VBPIndicator extends SMAIndicator {
             rangeStep: number,
             zoneStartsLength: number;
 
-        if (!lowRange || !highRange) {
+        // If the compare mode is set on the main series, change the VBP
+        // zones to fit new extremes, #16277.
+        const mainSeries = indicator.linkedParent;
+        if (
+            !indicator.options.compareToMain &&
+            mainSeries.options.compare &&
```
Comment (karolkolodziej) [inline ts/Stock/Indicators/VBP/VBPIndicator.ts:599]: Should this only be available for `compare`?

**Review by karolkolodziej — APPROVED** [review]
Two small questions.
Besides that all good!

**Review by pawellysy — APPROVED** [review]
Looks good!

File: ts/Stock/Indicators/VBP/VBPIndicator.ts [hunk 2]
```diff
@@ -590,7 +591,19 @@ class VBPIndicator extends SMAIndicator {
...
+            mainSeries.dataModify
+        ) {
+            lowRange = mainSeries.dataModify.modifyValue(lowRange) || 0;
```
[the remark this answers is not in the document]
Comment (raf18seb) [reply ts/Stock/Indicators/VBP/VBPIndicator.ts:602]: TS required it before the compare composition refactoring. You're right, after the merge it's not needed anymore.

File: ts/Stock/Indicators/VBP/VBPIndicator.ts (hunk 1 above)
Comment (raf18seb) [reply ts/Stock/Indicators/VBP/VBPIndicator.ts:599]: Again, some rubish before the compare composition improvements. Thanks!

**Review by pawelfus — APPROVED** [review]
Thanks!

**Review by TorsteinHonsi — APPROVED** [review]

**Pull request merged**